platform/mds/mds/src/main/java/org/motechproject/mds/config/MdsConfig.java

```diff
@@ -138,4 +147,16 @@ private void addBeanValidationFactoryProperty(Properties properties) {
         ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
         properties.put(PropertyNames.PROPERTY_VALIDATION_FACTORY, validatorFactory);
     }
+
+    private String getSchemaConnectionURL(Properties properties) {
+        String databaseName = properties.get(Constants.Util.SCHEMA_DATABASE_KEY) == null ? Constants.Util.DEFAULT_SCHEMA_DATABASE
+                : (String) properties.get(Constants.Util.SCHEMA_DATABASE_KEY);
+        return properties.get(CONNECTION_URL_KEY) + databaseName;
```

Comment:
what if connection_url doesn't end with a / ? Rather then doing this, we should have two full connection strings in datanucleus.properties, defaulting to ${sql.url}/db_name. This would allow database names to be configurable and would allow using different servers for the databases.

This class should have two methods available, that will return the current database names.

The default database name values should be placed in the default datanucleus.properties